hypothesis-python/src/hypothesis/strategies.py

```diff
@@ -333,6 +333,25 @@ def integers(min_value=None, max_value=None):
     min_int_value = None if min_value is None else ceil(min_value)
     max_int_value = None if max_value is None else floor(max_value)
 
+    if min_value != min_int_value and max_value != max_int_value:
```

Comment:
Instead of if/elif/elif, I would structure this as if/if - much easier to get full test coverage ([see failing coverage here](https://travis-ci.org/HypothesisWorks/hypothesis/jobs/439876841#L1093)), which still giving the warning if both are inexact.